### PR TITLE
Show status column with links and filter

### PR DESCRIPTION
**Pull request by brandones**
https://pihemr.atlassian.net/browse/CCI-70

screenshot--2020-04-20--134028

### CODE REVIEW

**Review by cioan — APPROVED** [review]
Looks good @brandones! Thanks.

File: src/declarations.d.tsx [hunk 1]
```diff
@@ -8,11 +8,13 @@ type Referral = {
...
-  patient_uuid: string;
   patient_name: string;
   referral_date: string;
   referral_type: string;
   status: string;
   visit_uuid: string;
   zl_emr_id: string;
+  fulfillment_status: string;
+  patient_uuid?: string;
+  person_uuid?: string;
```
Comment (mogoodrich) [inline src/declarations.d.tsx:19]: Do we still need person uuid here?

File: src/referrals-queue/referrals-queue.component.tsx [hunk 2]
```diff
@@ -117,6 +122,26 @@ export default function ReferralsQueue(props: ReferralsQueueProps) {
...
+              <label htmlFor="status">
+                <Trans i18nKey="status">Status</Trans>
+              </label>
+              <select
+                id="status"
+                value={statusFilter}
+                onChange={e => setStatusFilter(e.target.value)}
+                className={styles.dropdown}
+              >
+                <option value="">Any</option>
```
Comment (mogoodrich) [inline src/referrals-queue/referrals-queue.component.tsx:136]: Translation for "Any"?

File: src/root.component.tsx [hunk 3]
```diff
@@ -37,6 +37,23 @@ defineConfigSchema("@pih/esm-referrals-queue", {
...
+      },
+      url: {
+        default:
+          "/htmlformentryui/htmlform/editHtmlFormWithStandardUi.page?patientId=${patientUuid}&visitId=${visitUuid}&encounterId=${encounterUuid}&definitionUiResource=file:configuration/pih/htmlforms/section-mch-referral.xml",
+        validators: [validators.isString, validateUrlTemplate]
+      }
+    }
+  },
+  pendingStatuses: {
+    default: ["Pending status", "Referral unmet"],
```
Comment (mogoodrich) [inline src/root.component.tsx:54]: We are likely going to have to deal with localizing these?  I forget if the report returns localized statuses or not, could you check when you get a chance?

**Review by mogoodrich — APPROVED** [review]
Added a few comments, but generally LGTM! Thanks!

File: src/declarations.d.tsx (hunk 1 above)
Comment (brandones) [reply src/declarations.d.tsx:19]: I don't want to have to keep changing it back and forth, so I just made it compatible with both

File: src/referrals-queue/referrals-queue.component.tsx (hunk 2 above)
Comment (brandones) [reply src/referrals-queue/referrals-queue.component.tsx:136]: Thank you for calling that out! But in this case, I wasn't able to make it work the usual way, with the `<Trans>` component. So I put it off for later.

File: src/root.component.tsx (hunk 3 above)
Comment (brandones) [reply src/root.component.tsx:54]: Yes, that's the case. Maybe when it's time to do that is when I'll introduce Dave to the config file, and he can start managing some of these things.

File: src/referrals-queue/referrals-queue.component.tsx (hunk 2 above)
Comment (mogoodrich) [reply src/referrals-queue/referrals-queue.component.tsx:136]: Hmm, I guess there's no universal symbol for "any" we could use? :)

Comment (brandones) [reply src/referrals-queue/referrals-queue.component.tsx:136]: :shrug:

**Pull request merged**